CMakeLists.txt

```diff
@@ -111,9 +122,31 @@ if(USE_RTAUDIO)
   if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
     target_compile_definitions(RtAudio PUBLIC "__LINUX_ALSA__")
     target_link_libraries(RtAudio PUBLIC "asound")
+    if(JACK_FOUND)
+      target_compile_definitions(RtAudio PUBLIC "__UNIX_JACK__")
+      target_include_directories(RtAudio PUBLIC ${JACK_INCLUDE_DIRS})
+      link_directories(${JACK_LIBRARY_DIRS})
+      target_link_libraries(RtAudio PUBLIC ${JACK_LIBRARIES})
+    endif()
+    if(PULSEAUDIO_FOUND)
+      target_compile_definitions(RtAudio PUBLIC "__LINUX_PULSE__")
+      target_include_directories(RtAudio PUBLIC ${PULSEAUDIO_INCLUDE_DIRS})
+      link_directories(${PULSEAUDIO_LIBRARY_DIRS})
+      target_link_libraries(RtAudio PUBLIC ${PULSEAUDIO_LIBRARIES})
+    endif()
   elseif(CMAKE_SYSTEM_NAME STREQUAL "Windows")
     target_compile_definitions(RtAudio PUBLIC "__WINDOWS_DS__")
     target_link_libraries(RtAudio PUBLIC "dsound" "ole32")
+    target_compile_definitions(RtAudio PUBLIC "__WINDOWS_WASAPI__")
```

Comment:
Be careful!

Support for WASAPI will break Windows XP support!
Also, WASAPI thing is bad-met by professionals and there are prefer ASIO thing.

Reply:
The only thing the WASAPI is dynamically looking up via LoadLibrary() thing...

Reply:
ASIO is higher in the priority list and is taken if it's there, otherwise not.
> will break Windows XP support!
How can you be sure? yes it's Vista API, but I believe it's COM so it will just fail to instantiate it and go on to the next. But it's just hypothesis of me, and it's a good idea to test.

Reply:
Let's make a XP install to check this. It interests me because I have released windows adljack with WASAPI support compiled.

Reply:
> How can you be sure? yes it's Vista API, but I believe it's COM so it will just fail to instantiate it and go on to the next. But it's just hypothesis of me, and it's a good idea to test.

SDL2 had implemented WASAPI support, but with that my apps are won't work on Windows XP until I forcedly disabled WASAPI from SDL2 build, and when SDL2 implemented dynamical usage of WASAPI via LoadLibrary to keep support for WinXP

Reply:
Right I'll check this on XP before I go to sleep.
If it's a failure I'll remove it and I'll submit the dynamic loading to RtAudio upstream.